src/main.js

```diff
@@ -383,60 +381,31 @@ function handleAppWebContentsCreated(dc, contents) {
 
   contents.on('will-navigate', (event, url) => {
     const contentID = event.sender.id;
-    const parsedUrl = new URL(url);
-    const urlIsTrusted = isTrustedURL(parsedUrl);
-    const urlIsTrustedExternalLoginPath = parsedUrl.hostname === customLogins[contentID].externalHostname;
+    const parsedURL = parseURL(url);
+    const urlIsTrusted = isTrustedURL(parsedURL);
 
     // don't prevent custom login attempts (oath, saml)
```

Comment:
nit: the double negatives here are hard to parse. Perhaps something like:
```js
    if (urlIsTrusted) {
        return;
    }

    // allow arbitrary navigation while in a trusted login flow
    if (customLogins[contentID].inProgress) {
       return;
    }

    // Otherwise, prevent the webview from navigating outside the trusted domain. This
    // prevents security issues such as...
    event.preventDefault();
```

Reply:
I like this approach, just have to keep reminding myself to break my old habits.